### ISSUE TITLE
Use GitHub Actions matrices to build Docker images

### ISSUE THREAD

**Issue by tetsuo-cpp** [trailofbits/sigstore-conformance#18]
At the moment, we list each client/release channel combination in `ci.yml` to build each image. We should use the workflows' builtin support for test matrices to achieve this.

Waiting on #1.

**Comment by apstickler** [comment]
No longer needed with clarified direction from Customer.
